crypto/src/note.rs

```diff
@@ -204,3 +340,42 @@ impl TryFrom<&[u8]> for Commitment {
         Ok(Commitment(inner))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::keys::SpendKey;
+    use ark_ff::UniformRand;
+    use rand_core::OsRng;
+
+    #[test]
+    fn test_note_encryption_and_decryption() {
+        let mut rng = OsRng;
+
+        let sk = SpendKey::generate(&mut rng);
+        let fvk = sk.full_viewing_key();
+        let ivk = fvk.incoming();
+        let (dest, _dtk_d) = ivk.payment_address(0u64.into());
+
+        let value = Value {
+            amount: 10,
+            asset_id: b"pen".as_ref().into(),
+        };
+        let note = Note::new(
+            *dest.diversifier(),
+            dest.transmission_key(),
+            value,
+            Fq::rand(&mut rng),
+        )
+        .expect("can create note");
+        let esk = ka::Secret::new(&mut rng);
+
+        let ciphertext = note.encrypt(&esk).expect("can encrypt note");
+
+        let epk = esk.diversified_public(&dest.diversified_generator());
+        let plaintext = Note::decrypt(ciphertext, ivk, &epk).expect("can decrypt note");
+
+        assert_eq!(plaintext, note);
+    }
```

Comment:
In this test I think it'd be useful to generate `ivk2` (from a totally unrelated spend key) and check that decrypting the note with the wrong IVK gives an error.

Reply:
(motivation: then we know that we can use note decryption for scanning)